### ISSUE TITLE
Certain models have incorrect normals

### ISSUE THREAD

**Issue by BagelOrb** [Ultimaker/Uranium#34]
Some models show wrong overhangs.

Perhaps the normals as described in the STL file are incorrect.

It would then be nice to recalculate the normals so that the right pieces are colored red.
![pluggcura_wrong_overhangs](https://cloud.githubusercontent.com/assets/8895761/7412532/141330f2-ef44-11e4-98c5-c38ec13dbe20.png)

**Comment by BagelOrb** [comment]
The direction of a face is implicit in the winding order of its vertices. The normal is pointing upward from the plane in which the vertices are ordered counter-clockwise. As such we can just see whether two faces are flipped the same way by checking the order on the vertices; when two connected faces have their edges in the opposite direction they are correctly flipped.
![uitlegfacedirection](https://cloud.githubusercontent.com/assets/8895761/7678552/642651b8-fd55-11e4-8b4b-889dbfe0194f.png)

@Ghostkeeper your algorithm is superfluous.

**Comment by Ghostkeeper** [comment]
@BagelOrb Yeah that seems like a much more elegant solution than my overthinking, if the normals are indeed implicit in the winding order. It probably works better with edge cases as well.

**Comment by daid** [comment]
```
difference() { cube(20, center=true); cube(10, center=true); }
```
OpenSCAD code.

Result is a cube with a hollow center. But the center has an improper winding order.

Good luck ;-)

**Comment by BagelOrb** [comment]
The center hole cannot be seen, so it's not problem.

**Comment by awhiemstra** [comment]
Referenced the wrong commit, this shouldn't be closed yet.

**Comment by fieldOfView** [comment]
MeshTools plugin has a fairly robust option to fix normals.
